### ISSUE TITLE
Loader API onDuplicateKeyIgnore() doesn't work when primary key is an identity

### ISSUE THREAD

**Issue by MatiasSMd** [jOOQ/jOOQ#5200]
# Context

JOOQ 3.7.3
PostgreSQL 9.5.2
# Summary

Doing

``` java
dsl
  .loadInto(tableToLoad)
  .onDuplicateKeyIgnore()
  .loadCSV(file)
  .fields(fieldsNotContainingPK)
  .execute();
```

When the `file`contains many (valid) rows and where `fieldsNotContainingPK` are all the fields on the table `tableToLoad` except for the PK (which, in my case, is a serial/auto-increment), **only the first row in the file is saved into the DB**.
# Code analysis

The problem seems to be in `org.jooq.impl.LoaderImpl#executeSQL`. When solving the `onDuplicateKeyIgnore` logic, the code looks for the PK to create a select (where) condition. Not finding any PK field, the select is created without condition and every subsequent execution returns a result (the first row inserted) preventing any new row from being added. 
# Solution analysis

In my case, because I'm using Postgre 9.5, a "simple" solution would be to use the upsert logic built in in the dialect. This would do the correct thing and avoid this bug. 

Otherwise an optimistic approach may be useful (try to insert and, if fails due to a constraint), ignore the failure.

**Comment by lukaseder** [comment]
Thank you very much for your detailed analysis. This is very helpful. Indeed, we haven't covered the case where the primary key is not part of the field set. On the other hand, you might simply leave the `onDuplicateKeyIgnore` setting away in this case, and everything should work fine, right?

**Comment by MatiasSMd** [comment]
In my case I have a UNIQUE constraint that prevents me from inserting despite of not conflicting on the PK, so it is not possible to just skip the conflict resolution.

**Comment by lukaseder** [comment]
Oh, I see. Thanks for the update. Good to know about a use-case for this issue.

**Comment by lukaseder** [comment]
Finally looking into this now. This is quite easy to reproduce. There are two issues here:

1. Indeed, if we don't supply the primary key in the inserted values, then `ON DUPLICATE KEY IGNORE` should probably not consider it, if it is auto generated. This might not be a problem strictly related to the `Loader` API, will investigate this furter
2. We're still not using the `INSERT .. ON DUPLICATE KEY IGNORE` API, but run a `SELECT` query prior to the insertion, with all the drawbacks.

I will investigate this further and update this issue

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.14.0. Backport scheduled for 3.13.3 (#10357)
